lib/Resque/Job/Status.php

```diff
@@ -39,7 +39,14 @@ class Resque_Job_Status
 	 */
 	public function __construct($id)
 	{
-		$this->id = $id;
+		$this->id = self::generateId($id);
+	}
+
+	/**
+	 * generate job id consistently
+	 */
+	private static function generateId($id) {
```

Comment:
Using `private` visibility means we can't just inherit the class because public methods use private methods. How about `protected` visibility?